src/librustc_mir/transform/simplify.rs

```diff
@@ -331,8 +331,9 @@ struct DeclMarker {
 
 impl<'tcx> Visitor<'tcx> for DeclMarker {
     fn visit_local(&mut self, local: &Local, ctx: PlaceContext<'tcx>, _: Location) {
-        // ignore these altogether, they get removed along with their otherwise unused decls.
-        if ctx != PlaceContext::StorageLive && ctx != PlaceContext::StorageDead {
+        // ignore storage markers altogether, they get removed along with their otherwise unused
+        // decls.
+        if !ctx.is_storage_marker() {
```

Comment:
leave a FIXME that we'll want to extend this to all non-uses

Reply:
Fixed.